### PR TITLE
Make minor fixes in Compiler to properly handle void type expression

### PR DESCRIPTION
**Pull request by daxian-dbw**
## PR Summary
Make some minor fixes in Compiler to properly handle `void` type expressions.

### Before Fix
```
> [void]1,2,3
An error occurred while creating the pipeline.
+ CategoryInfo          : NotSpecified: (:) [], ParentContainsErrorRecordException
+ FullyQualifiedErrorId : RuntimeException
```
```
> [void]1 | Measure-Object
An error occurred while creating the pipeline.
+ CategoryInfo          : NotSpecified: (:) [], ParentContainsErrorRecordException
+ FullyQualifiedErrorId : RuntimeException
```
```
## The generated code for `[void](mkdir c:\test)` is discarded
> $s = @([void](mkdir c:\test))
> Test-Path c:\test
False
```

### After Fix
```
> $a = [void]1,2,3
> $a.Count
3
```
```
> [void]1 | Measure-Object

Count    : 0
Average  :
Sum      :
Maximum  :
Minimum  :
Property :
```
```
> $s = @([void](mkdir c:\test))
> $s.Count
0
> Test-Path C:\test
True
```

## PR Checklist

Note: Please mark anything not applicable to this PR `NA`.

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - [x] Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] User facing [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - [ ] Issue filed - Issue link:
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [x] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
    - [x] [Add `[feature]` if the change is significant or affectes feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready.

### CODE REVIEW

File: test/powershell/Language/Scripting/Array.Tests.ps1 [hunk 1]
```diff
@@ -72,4 +72,36 @@ Describe "ArrayExpression Tests" -Tags "CI" {
         $result.Length | Should Be 1
         $result[0] | Should Be $null
     }
+
+    It "@([void](New-Item)) should create file" {
+        try {
+            $testFile = "$TestDrive\test.txt"
```
Comment (iSazonov) [inline test/powershell/Language/Scripting/Array.Tests.ps1]: I'd like to see more reliable tests - could we use the file name based on guid or do `$testFile | Should Not Exist`?

**Review by iSazonov — APPROVED** [review]
Leave a comment

File: test/powershell/Language/Scripting/Array.Tests.ps1 (hunk 1 above)
Comment (lzybkr) [reply test/powershell/Language/Scripting/Array.Tests.ps1]: Isn't the contract with Pester to create a new directory for every `It` block? Assuming that's the case, reliability shouldn't be a concern.

**Review by lzybkr — APPROVED** [review]
You might consider testing the cast to `[void]` when it's not the first element, or all elements.

You might also consider testing a non-void method invocation cast to `void` as well as a `void` returning method invocation.

File: test/powershell/Language/Scripting/Array.Tests.ps1 (hunk 1 above)
Comment (iSazonov) [reply test/powershell/Language/Scripting/Array.Tests.ps1]: When I see tons of files and directories in my temp folder I always wonder this though a debugger and Ctrl-C are the simplest ways to create trash. :-)
From docs:
- TestDrive is a PowerShell PSDrive for file activity limited to the scope of a single Describe or Context block.

**Comment by iSazonov** [comment]
@daxian-dbw Is the PR ready to merge or you will continue?

**Comment by daxian-dbw** [comment]
@iSazonov Sorry for the late response. I will address Jason's and your comments.

**Comment by daxian-dbw** [comment]
@iSazonov @lzybkr Your comments have been addressed. Thank you!

**Comment by iSazonov** [comment]
LGTM.

**Pull request merged**